src/linker/Linker/DependencyInfo.cs

```diff
@@ -0,0 +1,142 @@
+		// used for security attributes, where we mark the type/properties directly
+		AttributeType, // attribute -> attribute type
+		AttributeProperty, // attribute -> attribute ptroperty
+		CustomAttributeArgumentType,
+		CustomAttributeArgumentValue,
+		CustomAttributeField,
+
+	// Tracking cctors
+		TriggersCctorThroughFieldAccess, // field-accessing method -> cctor of field's declaring type
+		TriggersCctorForCalledMethod, // caller method -> callee method type cctor
+		DeclaringTypeOfCalledMethod, // called method -> its declaring type (used to track when a cctor is triggered by a method call)
+		CctorForType, // type -> cctor of type
+		CctorForField, // field -> cctor of field's declaring type
+
+	// Tracking instantiations
+		InstantiatedByCtor, // ctor -> type it instantiates			
+		OverrideOnInstantiatedType, // instantiated type -> override method on the type
+
+	// Linker-specific behavior (preservation hints, patterns, user inputs, linker outputs, etc.)
+		PreservedDependency, // PreserveDependency attribute -> member
+		AccessedViaReflection, // method -> detected member accessed via reflection from that method
+		PreservedMethod, // type/method -> preserved method (explicitly preserved in Annotations by XML or other steps)
+		TypePreserve, // type -> field/method preserved for the type (explicitly set in Annotations by XML or other steps)
+		DisablePrivateReflection, // type/method -> DisablePrivateReflection attribute added by linkerf
+
+	// Built-in knowledge of special runtime/diagnostic subsystems
+		// XmlSchemaProvider, DebuggerDisplay, DebuggerTypeProxy, SoapHeader, TypeDescriptionProvider
+		ReferencedBySpecialAttribute, // attribute -> referenced members
+		KeptForSpecialAttribute, // attribute -> kept members (used when the members are not explicitly referenced)
+		SerializationMethodForType, // type -> method required for serialization
+		EventSourceProviderField, // EventSource derived type -> fields on nested Keywords/Tasks/Opcodes provider classes
+		MethodForSpecialType, // type -> methods kept (currently used for MulticastDelegate)
+
+	// Linker internals, requirements for certain optimizations
+		UnreachableBodyRequirement, // method -> well-known type required for unreachable bodies optimization
+		DisablePrivateReflectionRequirement, // null -> DisablePrivateReflectionAttribute type/methods (note that no specific source is reported)
+		AlreadyMarked, // null -> member that has already been marked for a particular reason (used to propagate reasons internally, not reported)
+
+	// For use when we don't care about tracking a particular dependency
+		Unspecified, // currently unused
+	}
+
+	readonly public struct DependencyInfo : IEquatable<DependencyInfo> {
+		public DependencyKind Kind { get; }
+		public object Source { get; }
+		public DependencyInfo (DependencyKind kind, object source) => (Kind, Source) = (kind, source);
+		public DependencyInfo (DependencyKind kind) => (Kind, Source) = (kind, default);
```

Comment:
This constructor does not make much sense to me (it should be set of static fields/properties instead)

Reply:
I'm not sure I understand. How would static fields work? I avoided setters so that it could be readonly.

Reply:
```C#
public static readonly DependencyInfo Unspecified = new DependencyInfo (DependencyKind.Unspecified, null);
```

Reply:
Gotcha, thanks!